src/cargo/ops/registry.rs

```diff
@@ -984,6 +984,14 @@ pub fn search(
         )
     })?;
 
+    if total_crates == 0 {
+        return Err(anyhow!(
```

Comment:
The motivation for this is scripting but we don't provide any programmatic output which makes me feel like we are providing a false promise by improving scripting support.

Reply:
I think the programmatic output is the error code in this case. It should be fairly trivial to check what the code is in most scripting languages.

Reply:
This is speculation.  I'm concerned that there is also parsing of the output (why do a `cargo search` and only care about the exit code?).  We should wait on an actual reply from the reporter